client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/step-content.tsx

```diff
@@ -9,11 +13,47 @@ type StepContentProps = {
 	goToStep?: NavigationControls[ 'goToStep' ];
 };
 
-const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => (
-	<div className="launchpad__content">
-		<Sidebar siteSlug={ siteSlug } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
-		<LaunchpadSitePreview siteSlug={ siteSlug } />
-	</div>
-);
+function sortByRegistrationDate( domainObjectA: ResponseDomain, domainObjectB: ResponseDomain ) {
+	const registrationDateA = domainObjectA.registrationDate;
+	const registrationDateB = domainObjectB.registrationDate;
+
+	if ( registrationDateA > registrationDateB ) {
+		return -1;
+	} else if ( registrationDateA < registrationDateB ) {
+		return 1;
+	}
+	return 1;
+}
+
+const StepContent = ( { submit, goNext, goToStep }: StepContentProps ) => {
+	const site = useSite();
+
+	const { data: allDomains = [] } = useGetDomainsQuery( site?.ID ?? null, {
+		retry: false,
+	} );
+
+	const domains = allDomains.map( createSiteDomainObject );
+
+	// Filter custom domains and sort by registration date.
+	// We want the first domain object to be the most recently registered domain
+	const nonWpcomDomains = domains
+		.filter( ( domain ) => ! domain.isWPCOMDomain )
+		.sort( sortByRegistrationDate );
+
+	const wpcomDomains = domains.filter( ( domain ) => domain.isWPCOMDomain );
+
+	const sidebarURL = nonWpcomDomains?.length
+		? nonWpcomDomains[ 0 ]?.domain
+		: wpcomDomains[ 0 ]?.domain;
+
+	const iFrameURL = wpcomDomains.length ? wpcomDomains[ 0 ]?.domain : nonWpcomDomains[ 0 ]?.domain;
```

Comment:
I'm wondering if there's ever a situation where we don't have access to a wpcom domain from Calypso 🤔